packages/evm/src/precompiles/bls12_381/noble.ts

```diff
@@ -0,0 +1,348 @@
+import {
+  BIGINT_0,
+  bigIntToBytes,
+  bytesToBigInt,
+  concatBytes,
+  equalsBytes,
+  setLengthLeft,
+} from '@ethereumjs/util'
+import { bls12_381 } from '@noble/curves/bls12-381'
+
+import { ERROR, EvmError } from '../../exceptions.js'
+
+import {
+  BLS_FIELD_MODULUS,
+  BLS_G1_INFINITY_POINT_BYTES,
+  BLS_G1_POINT_BYTE_LENGTH,
+  BLS_G2_INFINITY_POINT_BYTES,
+  BLS_G2_POINT_BYTE_LENGTH,
+  BLS_ONE_BUFFER,
+  BLS_ZERO_BUFFER,
+} from './constants.js'
+
+import type { EVMBLSInterface } from '../../types.js'
+import type { AffinePoint, ProjPointType } from '@noble/curves/abstract/weierstrass.js'
+
+// Copied from @noble/curves/bls12-381 (only local declaration)
+type Fp2 = {
+  c0: bigint
+  c1: bigint
+}
+
+/**
+ * Converts an Uint8Array to a MCL G1 point. Raises errors if the point is not on the curve
+ * and (if activated) if the point is in the subgroup / order check.
+ * @param input Input Uint8Array. Should be 128 bytes
+ * @param verifyOrder Perform the subgroup check (defaults to true)
+ * @returns MCL G1 point
+ */
+function BLS12_381_ToG1Point(input: Uint8Array) {
+  if (equalsBytes(input, BLS_G1_INFINITY_POINT_BYTES)) {
+    return bls12_381.G1.ProjectivePoint.ZERO
+  }
+
+  const x = bytesToBigInt(input.subarray(16, BLS_G1_POINT_BYTE_LENGTH / 2))
+  const y = bytesToBigInt(input.subarray(80, BLS_G1_POINT_BYTE_LENGTH))
+
+  const G1 = bls12_381.G1.ProjectivePoint.fromAffine({
+    x,
+    y,
+  })
+
+  return G1
+}
+
+// input: a mcl G1 point
```

Comment:
Do we need all these `mcl` references?  I'm assuming these are obsolete in this file, right?  Can they be replaced with something like `noble/curves`

Reply:
Don't think that a `noble/curves` reference is needed here, will just completely remove